### ISSUE TITLE
Errors when upgrading to terraform 0.13

### ISSUE THREAD

**Issue by llamahunter** [terraform-aws-modules/terraform-aws-notify-slack#109]
When trying to update to terraform 0.13 using terraform-aws-notify-slack v4.4.0, I am getting the following error during planning:

```
Error: Provider configuration not present

To work with
module.tivo_cluster.module.notify-slack.module.lambda.data.aws_caller_identity.current[0]
its original provider configuration at
module.tivo_cluster.provider["registry.terraform.io/-/aws"] is required, but
it has been removed. This occurs when a provider configuration is removed
while objects created by that provider still exist in the state. Re-add the
provider configuration to destroy
module.tivo_cluster.module.notify-slack.module.lambda.data.aws_caller_identity.current[0],
after which you can remove the provider configuration again.


Error: Provider configuration not present

To work with
module.tivo_cluster.module.notify-slack.module.lambda.data.aws_partition.current[0]
its original provider configuration at
module.tivo_cluster.provider["registry.terraform.io/-/aws"] is required, but
it has been removed. This occurs when a provider configuration is removed
while objects created by that provider still exist in the state. Re-add the
provider configuration to destroy
module.tivo_cluster.module.notify-slack.module.lambda.data.aws_partition.current[0],
after which you can remove the provider configuration again.


Error: Provider configuration not present

To work with
module.tivo_cluster.module.notify-slack.module.lambda.data.aws_region.current[0]
its original provider configuration at
module.tivo_cluster.provider["registry.terraform.io/-/aws"] is required, but
it has been removed. This occurs when a provider configuration is removed
while objects created by that provider still exist in the state. Re-add the
provider configuration to destroy
module.tivo_cluster.module.notify-slack.module.lambda.data.aws_region.current[0],
after which you can remove the provider configuration again.
```

Note that I have _already_ run `terraform 0.13upgrade` on my modules and deployment terraform. Looking at the 
notify-slack module, it appears you are not unisg explicit provider source locations in your code?  Please read
https://www.terraform.io/upgrade-guides/0-13.html#explicit-provider-source-locations

It appears you are still using 0.12 format?  e.g.

```
  required_providers {
    aws = ">= 2.35, < 4.0"
  }
```

rather than 

```
  required_providers {
    aws = {
      source = "hashicorp/aws"
      version = ">= 2.35, < 4.0"
    }
  }
```

**Comment by llamahunter** [comment]
Possibly, this has to do with your change of the lambda submodule, which removed resources that were still in use?  From the terraform 0.13 upgrade guide linked above

If you remove a resource block (or a module block for a module that contains resource blocks) before the first terraform apply, you may see a message like this reflecting that Terraform cannot determine which provider configuration the existing object ought to be managed by:

```
Error: Provider configuration not present

To work with null_resource.foo its original provider configuration at
provider["registry.terraform.io/-/aws"] is required, but it has been removed.
This occurs when a provider configuration is removed while objects created by
that provider still exist in the state. Re-add the provider configuration to
destroy aws_instance.example, after which you can remove the provider
configuration again.
```

In this specific upgrade situation the problem is actually the missing resource block rather than the missing provider block: Terraform would normally refer to the configuration to see if this resource has an explicit provider argument that would override the default strategy for selecting a provider. If you see the above after upgrading, re-add the resource mentioned in the error message until you've completed the upgrade.

So, I think you need to add back the missing resources that were deleted?

**Comment by llamahunter** [comment]
Since these were just 'data' objects, not 'resources', I deleted them from the remote state using `terraform state rm`, and it seems to have unstuck me.  Still, there should probably be something in the README.md about how to make the transition from the 3.x to 4.x versions of this module.

**Comment by bryantbiggs** [comment]
which version of 0.13.x are you using? there were known state migration issues on the earlier versions of 0.13 (0.13.1-0.13.2). its not an issue with the module here, but more with terraform itself 
- https://github.com/hashicorp/terraform/blob/v0.13/CHANGELOG.md#0132-september-02-2020
- https://github.com/hashicorp/terraform/blob/v0.13/CHANGELOG.md#0132-september-02-2020

**Comment by llamahunter** [comment]
using terraform 0.13.4

**Comment by eric-johnson** [comment]
I had the same problem.
I started with a clean plan in Terraform 0.12.29 and module version 3.5 (and a clean git workspace)
I pinned the module version to 4.0.0, ran `terraform_0.13.5 0.13upgrade` and `terraform_0.13 init`. Both were clean.
I get the errors above upon running `terraform_0.13.5 plan`

So I stashed my changes, and using Terraform 0.12.29 did a `terrafrom init` && `terraform state rm` each of these three data objects. Then brought back my 0.13 changes, `terraform_0.13.5 init`, and was able to apply the changes between 3.5 and 4.0.
